presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1637,6 +1637,40 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             return orderByScope;
         }
 
+        private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope outputScope, List<FunctionCall> aggregations, List<List<Expression>> groupByExpressions)
+        {
+            // When aggregation is present then FROM (non-aggregated) fields should be hidden for ORDER BY during planning.
```

Comment:
The scope `computeAndAssignOrderByScopeWithAggregation` is only used in planning (not analysis), because planner requires scope in order to bind column names to fields. Different scope for ORDER BY analysis is created in `computeAndAssignOrderByScope`. There are tests for that.

This is another rough edge. When we move to scopeless planning, then this scope won't be needed at all. However, I don't want this PR to explode. I will add proper comments